### PR TITLE
audiounit: aggregate devices follow up

### PR DESCRIPTION
**Pull request by achronop**
The patch addresses the follow up comments of PR #231

### CODE REVIEW

File: src/cubeb_audiounit.cpp [hunk 1]
```diff
@@ -1451,12 +1406,7 @@ audiounit_create_blank_aggregate_device(AudioObjectID * plugin_id, AudioDeviceID
   CFDictionaryAddValue(aggregate_device_dict, CFSTR(kAudioAggregateDeviceIsPrivateKey), aggregate_device_private_key);
   CFRelease(aggregate_device_private_key);
 
-  r = AudioObjectGetPropertyData(*plugin_id,
-                                 &create_aggregate_device_address,
-                                 sizeof(aggregate_device_dict),
-                                 &aggregate_device_dict,
-                                 &size,
-                                 aggregate_device_id);
+  OSStatus r = AudioHardwareCreateAggregateDevice( aggregate_device_dict, aggregate_device_id);
```
Comment (kinetiknz) [inline src/cubeb_audiounit.cpp]: No space between ( and aggregate_device_dict

File: src/cubeb_audiounit.cpp [hunk 2]
```diff
@@ -1480,13 +1430,10 @@ get_device_name(AudioDeviceID id)
 }
 
 static int
-audiounit_set_aggregate_sub_device_list(cubeb_stream * stm)
+audiounit_set_aggregate_sub_device_list(AudioDeviceID aggregate_device_id, AudioDeviceID input_device_id,
+                                        AudioDeviceID output_device_id)
```
Comment (kinetiknz) [inline src/cubeb_audiounit.cpp]: When you have to wrap function params, it's more readable if you put one on each line.

**Comment by achronop** [comment]
Thanks for the review.

**Comment by achronop** [comment]
It does not build for OSX 10.7 (as expected) because AudioHardwareCreateAggregateDevice and AudioHardwareDestroyAggregateDevice do not exist there.
https://treeherder.mozilla.org/#/jobs?repo=try&revision=89cd68a8f11981106a12d0a13659a88694168afe

You mentioned we do not need to support the audiounit backend for versions earlier that 10.9. What do you suggest for those versions?

**Comment by kinetiknz** [comment]
It looks like our builders are still on 10.7 and building against the 10.7 API, despite Firefox supporting 10.9 as the minimum since 48 in August 2016.  I guess we'll have to use the old code paths until either bug 1270217 is fixed  or the builders are migrated to >= 10.9.

**Pull request merged**